src/main/kotlin/no/nav/familie/ks/sak/statistikk/saksstatistikk/SakStatistikkService.kt

```diff
@@ -4,40 +4,46 @@ import no.nav.familie.kontrakter.felles.objectMapper
 import no.nav.familie.ks.sak.api.dto.BehandlingPåVentResponsDto
 import no.nav.familie.ks.sak.kjerne.arbeidsfordeling.ArbeidsfordelingService
 import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingRepository
-import no.nav.familie.ks.sak.kjerne.behandling.domene.BehandlingStegTilstand
+import no.nav.familie.ks.sak.kjerne.behandling.steg.BehandlingSteg
 import no.nav.familie.ks.sak.kjerne.behandling.steg.BehandlingStegStatus
+import no.nav.familie.ks.sak.kjerne.fagsak.domene.FagsakRepository
 import no.nav.familie.ks.sak.kjerne.totrinnskontroll.TotrinnskontrollService
 import no.nav.familie.prosessering.domene.Task
 import no.nav.familie.prosessering.internal.TaskService
 import org.springframework.stereotype.Service
+import java.time.LocalDateTime
 import java.time.OffsetDateTime
 import java.time.ZoneOffset
 import java.util.Properties
 
 @Service
 class SakStatistikkService(
     private val behandlingRepository: BehandlingRepository,
+    private val fagsakRepository: FagsakRepository,
     private val taskService: TaskService,
     private val totrinnskontrollService: TotrinnskontrollService,
     private val arbeidsfordelingService: ArbeidsfordelingService
 ) {
 
-    fun opprettSendingAvBehandlingensTilstand(behandlingId: Long, stegTilstand: BehandlingStegTilstand) {
+    fun opprettSendingAvBehandlingensTilstand(behandlingId: Long, behandlingSteg: BehandlingSteg) {
+        val behandlingStegTilstand =
+            behandlingRepository.hentBehandling(behandlingId).behandlingStegTilstand.singleOrNull { it.behandlingSteg == behandlingSteg }
         val hendelsesbeskrivelse = "Ny behandlingsstegstilstand " +
-            "${stegTilstand.behandlingSteg}:${stegTilstand.behandlingStegStatus} " +
+            "${behandlingStegTilstand?.behandlingSteg}:${behandlingStegTilstand?.behandlingStegStatus} " +
             "for behandling $behandlingId"
 
         val tilstand = hentBehandlingensTilstand(behandlingId)
-        opprettProsessTask(behandlingId, tilstand, hendelsesbeskrivelse)
+        opprettProsessTask(behandlingId, tilstand, hendelsesbeskrivelse, SendBehandlinghendelseTilDvhTask.TASK_TYPE)
     }
 
     private fun opprettProsessTask(
         behandlingId: Long,
         behandlingTilstand: BehandlingStatistikkDto,
-        hendelsesbeskrivelse: String
+        hendelsesbeskrivelse: String,
+        type: String
     ) {
         val task = Task(
-            type = SendBehandlinghendelseTilDvhTask.TASK_TYPE,
+            type = type,
```

Comment:
Kommer vi til å opprette flere typer tasker her enn `SendBehandlinghendelseTilDvhTask.TASK_TYPE` her? Kunne isåfall vurdert å fjerne den som parameter.

Reply:
Vi har 2 tasktyper, en for sending av enkelthendelser og en for sending av alle behandlinger, derfor typen som parameter